ql/src/java/org/apache/hadoop/hive/ql/ddl/view/materialized/alter/rebuild/AlterMaterializedViewRebuildAnalyzer.java

```diff
@@ -69,6 +72,23 @@ public void analyzeInternal(ASTNode root) throws SemanticException {
 
     LOG.debug("Rebuilding materialized view " + tableName.getNotEmptyDbTable());
     super.analyzeInternal(rewrittenAST);
+
+    try {
+      Table table = db.getTable(tableName.getDb(), tableName.getTable());
```

Comment:
Is there any reason not to do this check right after tableName is parsed from the tree? Probably I'm missing something, but I believe that right after  line 62 this could be checked - no need to create the rootTasks then clear them. Also it would be nice to move this logic into a separate function, returning. boolean if it is outdated or not.

Reply:
Thanks for highlighting that we have all necessary info before calling `super.analyzeInternal()`.
Creating `HiveOperation.ALTER_MATERIALIZED_VIEW_REBUILD` was necessary to get this work but no need to modify the `rootTasks`.